### PR TITLE
fixed modulo arithmetic for previous song

### PR DESCRIPTION
**Pull request by Ben-Graham**

### CODE REVIEW

**Comment by bill-luu** [comment]
Can you explain how this works again?

**Comment by ExtraGravity** [comment]
That comment probably means another change needs to happen LOL. Either expand it out to be more readable or a comment to explain it.

**Comment by Ben-Graham** [comment]
In C++ the % operator is the modulo operator for positive numbers. When working with negative numbers which sometimes happens when going to a previous song, the % works as the remainder operator which gives different results.  

An in depth explanation with examples can be found [here.](https://stackoverflow.com/questions/7594508/modulo-operator-with-negative-values)

File: BusinessLayer/SongControl/SongControl.cpp [hunk 1]
```diff
@@ -21,7 +21,7 @@ QString SongControl::nextSong()
 
 QString SongControl::previousSong()
 {
-    currentSongIndex_ = (currentSongIndex_ - 1) % files_.size();
+    currentSongIndex_ = (((currentSongIndex_ - 1) % files_.size())+files_.size()) % files_.size();
```
Comment (bseto) [inline BusinessLayer/SongControl/SongControl.cpp]: I think for readability, it'd just be easier if you did an `if` statement.
```
currentSongIndex_ -= 1;
if (currentSongIndex_ < 0) {
    currentSongIndex_ = files_.size() - 1;
}
```

Note: For future readers. C++ does not wrap negatives on modulus - I think...
https://stackoverflow.com/a/42964502

**Review by ExtraGravity — APPROVED** [review]

**Comment by ExtraGravity** [comment]
Wait, this didn't even compile. The travis failed.

**Comment by bseto** [comment]
Should the settings be changed to ensure passing Travis before allowing a merge?

**Comment by ExtraGravity** [comment]
Yes. haha, this was a mistake

**Pull request merged**